sbin/submitty_daemon_jobs/submitty_jobs/generate_pdf_images.py

```diff
@@ -13,28 +13,55 @@ def __init__(self, page_number: int, coordinates: Sequence[float]):
         self.coordinates = coordinates
 
 
-def main(pdf_file_path: str, redactions: List[Redaction]):
+def main(pdf_file_path: str, output_dir: str, redactions: List[Redaction]):
     directory = os.path.dirname(pdf_file_path)
     if directory:
         os.chdir(os.path.dirname(pdf_file_path))
+    # Ensure the output directory exists
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+    # Delete any existing images that match the pattern
+    for filename in os.listdir(output_dir):
+        if filename.startswith(os.path.basename(pdf_file_path)[:-4]) and filename.endswith(".jpg"):
+            os.remove(os.path.join(output_dir, filename))
```

Comment:
This is pretty difficult to read, to the point that I'm not entirely sure whether it's correct to not.  Is there a cleaner way to accomplish the same thing?